src/telliot_feed_examples/reporters/rng_interval.py

```diff
@@ -0,0 +1,355 @@
+"""TellorRNG auto submitter.
+submits TellorRNG values at a fixed time interval
+"""
+import asyncio
+import calendar
+import time
+from datetime import timedelta
+from typing import Any
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import requests
+from chained_accounts import ChainedAccount
+from eth_utils import to_checksum_address
+from telliot_core.contract.contract import Contract
+from telliot_core.datafeed import DataFeed
+from telliot_core.model.endpoints import RPCEndpoint
+from telliot_core.queries.tellor_rng import TellorRNG
+from telliot_core.utils.log import get_logger
+from telliot_core.utils.response import error_status
+from telliot_core.utils.response import ResponseStatus
+
+from telliot_feed_examples.feeds.matic_usd_feed import matic_usd_median_feed
+from telliot_feed_examples.feeds.tellor_rng_feed import assemble_rng_datafeed
+from telliot_feed_examples.feeds.trb_usd_feed import trb_usd_median_feed
+from telliot_feed_examples.reporters.interval import IntervalReporter
+from telliot_feed_examples.reporters.reporter_autopay_utils import get_feed_tip
+from telliot_feed_examples.reporters.reporter_autopay_utils import get_single_tip
+
+
+logger = get_logger(__name__)
+
+INTERVAL = 60 * 30  # 30 minutes
+START_TIME = 1653350400  # 2022-5-24 00:00:00 GMT
+
+
+def get_next_timestamp() -> int:
+    """get next target timestamp"""
+    now = calendar.timegm(time.gmtime())
+    target_ts = START_TIME + (now - START_TIME) // INTERVAL * INTERVAL
+    return target_ts
+
+
+class RNGReporter(IntervalReporter):
+    """Reports TellorRNG values at a fixed interval to TellorFlex
+    on Polygon."""
+
+    def __init__(
+        self,
+        endpoint: RPCEndpoint,
+        account: ChainedAccount,
+        chain_id: int,
+        oracle: Contract,
+        token: Contract,
+        autopay: Contract,
+        stake: float = 10.0,
+        datafeed: Optional[DataFeed[Any]] = None,
```

Comment:
I think you want to inherit from `PolygonReporter` instead of `IntervalReporter`, no? Also, leave out everything from the constructor except for overwriting the datafeed with an RNG datafeed like how you're doing in `report_once()`:
```python
self.datafeed = await assemble_rng_datafeed(
            timestamp=rng_timestamp, node=self.endpoint, account=self.account
        )
```

You also don't have to include any of the class methods that you aren't changing (like `ensure_staked()` or `check_reporter_lock()`), since they're inherited.